services/emergency-room-svc/main.go

```diff
@@ -216,6 +217,85 @@ func whereClausesForERsQuery(db *gorm.DB, request *api.GetERsRequestV1) *gorm.DB
 	}
 	return db
 }
+func updateERHandler(ctx context.Context, in *daprcmn.InvocationEvent) (*daprcmn.Content, error) {
+	log, logCtx := common.GetHandlerLogger("updateERHandler", ctx)
+
+	// TODO: Auth
+
+	// Parse
+	request := api.UpdateERRequestV1{}
+	if err := hwutil.ParseValidJson(in.Data, &request); err != nil {
+		log.Warn().Err(err).Msg("invalid input")
+		return nil, err
+	}
+	log.Debug().Str("body", logging.Formatted(request)).Send()
+
+	// Update
+	db := hwgorm.GetDB(logCtx)
+
+	emergencyRoom := models.EmergencyRoom{
+		ID: request.ID,
+	}
+
+	updateMap := updatesMap(&request)
```

Comment:
Can we "map" the UpdateERRequestV1 struct to an EmergencyRoom struct and use this for the update instead of a map?

Reply:
no because of go's handling of default values. e.g. updating open to "false" will not do anything as gorm thinks it's a default value and ignore it

Reply:
Okay, i guess then we should add the updatesMap function as a method to the UpdateERRequestV1 struct.